pi-blaster.c

```diff
@@ -70,23 +72,32 @@ static uint8_t pin2gpio[8];
 // will use too much memory bandwidth.  10us is a good value, though you
 // might be ok setting it as low as 2us.
 
-#define CYCLE_TIME_US		10000
+#define CYCLE_TIME_US	10000
 #define SAMPLE_US		10
 #define NUM_SAMPLES		(CYCLE_TIME_US/SAMPLE_US)
 #define NUM_CBS			(NUM_SAMPLES*2)
 
-#define NUM_PAGES		((NUM_CBS * 32 + NUM_SAMPLES * 4 + \
+#define NUM_PAGES		((NUM_CBS * sizeof(dma_cb_t) + NUM_SAMPLES * 4 + \
 					PAGE_SIZE - 1) >> PAGE_SHIFT)
 
-#define DMA_BASE		0x20007000
-#define DMA_LEN			0x24
-#define PWM_BASE		0x2020C000
+#define DMA_BASE		(periph_virt_base + 0x00007000)
+#define DMA_CHAN_SIZE	0x100 /* size of register space for a single DMA channel */
+#define DMA_CHAN_MAX	14  /* number of DMA Channels we have... actually, there are 15... but channel fifteen is mapped at a different DMA_BASE, so we leave that one alone */
+#define DMA_CHAN_NUM	14  /* the DMA Channel we are using, NOTE: DMA Ch 0 seems to be used by X... better not use it ;) */
```

Comment:
I tested the `/sys/class/dma/<channel>/in_use` interface:

```shell
~/pi-blaster $ ll /sys/class/dma/dma0chan? -d
lrwxrwxrwx 1 root root 0 Apr 15 06:43 /sys/class/dma/dma0chan0 -> ../../devices/platform/bcm2708-dmaengine/dma/dma0chan0
lrwxrwxrwx 1 root root 0 Apr 15 06:43 /sys/class/dma/dma0chan1 -> ../../devices/platform/bcm2708-dmaengine/dma/dma0chan1
lrwxrwxrwx 1 root root 0 Apr 15 06:43 /sys/class/dma/dma0chan2 -> ../../devices/platform/bcm2708-dmaengine/dma/dma0chan2
lrwxrwxrwx 1 root root 0 Apr 15 06:43 /sys/class/dma/dma0chan3 -> ../../devices/platform/bcm2708-dmaengine/dma/dma0chan3
lrwxrwxrwx 1 root root 0 Apr 15 06:43 /sys/class/dma/dma0chan4 -> ../../devices/platform/bcm2708-dmaengine/dma/dma0chan4
~/pi-blaster $ cat /sys/class/dma/dma0chan?/in_use
1
1
0
0
0
```
So we can see, that only 4 channels are "available" in this interface... i tested using DMA Channel 4 in pi-blaster... if i use that channel in the `/sys/...` interface channel 4 is still marked as *not* in_use...  so i guess using that interface only makes sense if we can also request the dma channel from the kernel... as long as we cannot do that (waiting for an answer to your stackoverflow question) i think that hardcoding to channel 14 is the best alternative we have... if we can use the kernel to get a DMA Channel that would be much nicer (also for the PWM/PCM and other hardware we are using btw) ... but i guess that would be a different PR :)